src/pod.ts

```diff
@@ -122,15 +122,15 @@ export abstract class AbstractPod extends base.Resource implements IPodSelector,
   public addInitContainer(cont: container.ContainerProps): container.Container {
 
     // https://kubernetes.io/docs/concepts/workloads/pods/init-containers/#differences-from-regular-containers
-    if (cont.readiness) {
+    if (cont.restartPolicy != container.ContainerRestartPolicy.ALWAYS && cont.readiness) {
```

Comment:
So a container that has `ContainerRestartPolicy.ALWAYS` is considered a sidecar? Can you elaborate and refer to some docs explaining this? Also, lets put the new condition in a function:

```suggestion
    if (!this.isSidecar(cont) && cont.readiness) {
```

Reply:
We also need some tests for these new conditions

Reply:
Ah ok I see its explained in the issue: https://github.com/cdk8s-team/cdk8s-plus/issues/4338#issuecomment-2191018002.

I would still make a note on this in the code for future readers, and put it in a separate function so its more readable.

Reply:
> So a container that has ContainerRestartPolicy.ALWAYS is considered a sidecar? Can you elaborate and refer to some docs explaining this?

basically, yes. any init container having `ContainerRestartPolicy.ALWAYS` is a sidecar container. you can see the differences here: https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/#differences-from-init-containers

I see your concerns. I will add a new function to check if container is sidecar. Also I will add a comment with a link to documentation for future collaborators and add more tests and try to cover everything